File: testdata/lib-ts/d2.ts
import { Construct } from 'constructs';


export function  Show(c: Construct) {
        c.node.addMetadata("Show", "true")
}

export function  Connection(source: Construct, target: Construct) {
    source.node.addMetadata("Connection", target.node.id)
}
export function  Container(source: Construct, target: Construct) {
    source.node.addMetadata("Container", target.node.id)
}

File: testdata/lib-go/d2.go
package d2

import (
		"github.com/aws/constructs-go/constructs/v10"
		"github.com/aws/aws-sdk-go-v2/aws"
)

type NodeAble interface {
	Node() constructs.Node
}	

func Show( construct NodeAble){
	construct.Node().AddMetadata(aws.String("Show"), aws.String("true"),nil)
}

func Connection( construct NodeAble, target NodeAble){
	construct.Node().AddMetadata(aws.String("Connection"), target.Node().Id(),nil)
}